src/templates/singleResource.js

```diff
@@ -0,0 +1,227 @@
+
+    // URL is format youtu.be, we need to update that
+    if (alternateBaseUrlIndex >= 0) {
+      urlArr.splice(ytIndex, 1, "youtube.com")
+    }
+
+    // URL has /watch we need to remove that
+    if (watchModeIndex >= 0) {
+      urlArr.splice(watchModeIndex, 1)
+    }
+
+    if (ytIndex >= 0 && embedModeIndex === -1) {
+      // We need to add /embed
+      urlArr.splice(ytIndex + 1, 0, "embed")
+      return urlArr.join("/")
+    } else if (ytIndex && embedModeIndex >= 0) {
+      // Url is already in the correct format
+      return url
+    } else {
+      // url is invalid (not youtube or bad format)
+      return false
+    }
+  }
+
+  const buildGalleryImages = images => {
+    return images.map(image => {
+      const src = image.fixed ? image.fixed.src : null
+      const title = image.title ? image.title : null
+      const caption = image.description ? image.description : null
+
+      return {
+        src,
+        title,
+        caption,
+      }
+    })
+  }
+
+  return (
+    <Layout active="single-resource" bgColor="magenta">
+      <SEO title="Resource" />
+      <HeaderArea
+        topLeft={returnHeaderLeft()}
+        topRight={width > 889 ? returnHeaderRight() : null}
+        customTop={true}
+      />
+      <div className="single-resource__container">
+        <SocialMediaBar title={title} url={url} hashtags={["arts-matter"]} />
+        {/* 
+              |     Move these sections to components later   |
+              V                                               V      */}
+        {classroomPhotos && (
+          <div className="single-resource__classroom-photos">
+            <ImageGallery images={buildGalleryImages(classroomPhotos)} />
+          </div>
+        )}
+        <div className="single-resource__videos">
+          {videos &&
+            videos.map((video, i) => {
+              const embedUrl = getEmbedUrl(video)
+
+              if (embedUrl !== false) {
+                return (
+                  <iframe
+                    title={i}
+                    className="single-resource__video"
+                    src={getEmbedUrl(video)}
+                    frameBorder="0"
+                    allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture fullscreen"
+                    allowFullScreen={true}
+                    webkitallowfullscreen="true"
+                    mozallowfullscreen="true"
+                    key={i}
+                  >Your browser doesn't support IFrame :/</iframe>
+                )
+              }
+
+            })}
+        </div>
+        {studentArtwork && (
+          <div className="single-resource__artwork">
+            <h4>Student Artwork</h4>
+            <div className="single-resource__artwork-wrapper">
+              {studentArtwork.map((artwork, i) => {
+                return (
+                  <div key={i} className="single-resource__artwork-container">
+                    <img
```

Comment:
Can we look into providing lightbox for this section? It would be nice to have a good way to view the larger version of images. Here are two resources that might be helpful:

https://www.gatsbyjs.com/blog/2018-11-03-building-an-accessible-lightbox/

https://reach.tech/dialog/ - this is the library leveraged in the example above.

Reply:
I think this would be a great addition! I'll look into adding that in.

Reply:
The lightbox wasn't too shabby to add! Much better with the ability to expand those images.